core/domain/topic_domain.py

```diff
@@ -562,21 +688,27 @@ def get_subtopic_index(self, subtopic_id):
                 return ind
         return None
 
-    def add_subtopic(self, subtopic_id, title):
+    def add_subtopic(self, title):
         """Adds a subtopic with the given id and title.
 
         Args:
-            subtopic_id: str. The id for the new subtopic.
             title: str. The title for the new subtopic.
 
         Raises:
             Exception. A subtopic with the given id already exists.
+
+        Returns:
+            SubtopicPage. A SubtopicPage object corresponding the newly created
+                subtopic.
         """
-        if self.get_subtopic_index(subtopic_id) is not None:
-            raise Exception(
-                'A subtopic with id %s already exists. ' % subtopic_id)
+        subtopic_page_id = self.get_new_subtopic_page_id()
+        subtopic_page = SubtopicPage(
```

Comment:
Hm, maybe return the subtopic_id and have the client do all the subtopic_page stuff instead? The latter seems out of scope for a method in Topic.

Reply:
Done.